log4j-api/pom.xml

```diff
@@ -28,7 +28,7 @@
   <name>Apache Log4j API</name>
   <description>The Apache Log4j API</description>
   <properties>
-    <javadoc.skip>false</javadoc.skip>
+    <maven.javadoc.skip>false</maven.javadoc.skip>
```

Comment:
This PR also replaces our custom `javadoc.skip` property with the official one.

Reply:
Yes, and this is in line with `main` too. But, what do you mean with your remark? Is this a problem and should be reverted? If so, why exactly?

Reply:
No, it is just a note for future readers if Git blame points them to this PR.